### DISCUSSION TITLE
Why can I only connect via USB and not Radio?

### DISCUSSION THREAD

**Discussion by buabula** [bitcraze/discussions#1387] [Q&A]
Hello,

I'm having trouble scanning and connecting my Bolt to my PC, though the USB connection works fine. I've checked the radio address and channel, and both are set to the default settings.

One thing to note is that sometimes I can detect a radio signal, but it only shows channel "74" or "73", and I'm unable to connect to them. When I change the channel via USB to a smaller number, the channel I detect also changes correspondingly. For example, if I set the channel to "74", I will scan "radio://0/66/2M" in the "Select an interface" section. Unfortunately, I still cannot connect.

Here is some related information:
![1](https://github.com/bitcraze/discussions/assets/113815745/5c0df2ab-d179-4b5e-9ede-1b66d540275c)
![2](https://github.com/bitcraze/discussions/assets/113815745/eb7929b0-8d70-4d09-b38f-31bc87991a8e)
![3](https://github.com/bitcraze/discussions/assets/113815745/59578243-f10e-4d9a-a99a-cb886f408751)
![5](https://github.com/bitcraze/discussions/assets/113815745/8360de69-50f7-4aa8-96bd-44725e19bbc0)
![6](https://github.com/bitcraze/discussions/assets/113815745/795f44f8-a859-48c6-af11-b0cbaf2bb87e)

Thank you for any help!!!

**Comment by ToveRumar** [comment]
Hi!
Ok, this looks a bit strange actually, that it changes channel with an offset.

Does the bolt have default FW loaded on it, or did you flash the latest release? 

Just to make sure, are you resetting the device between changing the radio channel and scanning+connecting? 

Is it possible for you to connect to it over radio when in cold boot mode (Bootloader->Cold Boot(recovery)). In cold boot it will always be on default address (ch.80 and default address 0xE7E7E7E7E7).

**Reply by buabula** [reply]
Thanks for you reply!! 

Firstly, I flashed the latest release STM32 2023.11, but I cannot download the source code directly by"$ git clone --recursive https://github.com/bitcraze/crazyflie-firmware.git" :
![image](https://github.com/bitcraze/discussions/assets/113815745/db065071-8069-4337-8e14-d7f200f5f299)
So, I download the missing files manually, however, the log shown “modified” at the beginning.
![image](https://github.com/bitcraze/discussions/assets/113815745/88677957-5f8d-416e-836f-4fdf94400ca4)
 
Secondly, I am sure I haven't reset the device in the process. If you mean flashing something. In fact I haven't do anything between changing and scanning.  And I even cannot stably scan for the radio channels.

Thirdly, I cannot connect to it over radio when in cold boot mode.

Thank you again!

**Reply by ToveRumar** [reply]
Hi!

> Firstly, I flashed the latest release STM32 2023.11, but I cannot download the source code directly by"$ git clone --recursive 

 It looks like many people are having this issues with git and openssl when using WSL2, are you using WSL? 

> Secondly, I am sure I haven't reset the device in the process. If you mean flashing something. In fact I haven't do anything between 
> changing and scanning. And I even cannot stably scan for the radio channels.

Ok, to start you can try restarting the module between changing the channel/address and scanning. You always need to reset the device for the changes to take effect. 

To rule out any FW issues, could you flash the latest released FW? I.e in the cfclient choose "From release" and the correct platform (cf2). This will flash the latest stable release. 
![Screenshot from 2024-06-20 10-42-24](https://github.com/bitcraze/discussions/assets/45783739/991da8c6-9128-4990-921f-ce94464542c1)


> Thirdly, I cannot connect to it over radio when in cold boot mode.

This is strange, and might be a radio issue. Are the blue leds blinking slowly, indicating cold boot mode? 
Does cfclient indicate that it does not find the device?
